### PR TITLE
Ensure pg_config --cppflags are passed

### PR DESCRIPTION
**Pull request by sumerman**
`CMAKE_CPP_FLAGS` is not a thing at all. Furthermore, `CMAKE_CXX_FLAGS` is not passed to a C compiler.
`pg_config` uses `CPPGLAGS` for all includes, and needs to be propagated into `CMAKE_C_FLAGS` as well.

I ran into #2777 on my system and that's what prompted this PR.

### CODE REVIEW

File: src/build-defs.cmake [hunk 1]
```diff
@@ -7,8 +7,8 @@ if(UNIX)
   set(CMAKE_C_STANDARD 11)
   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -L${PG_LIBDIR}")
   set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -L${PG_LIBDIR}")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${PG_CFLAGS}")
-  set(CMAKE_CPP_FLAGS "${CMAKE_CPP_FLAGS} ${PG_CPPFLAGS}")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${PG_CFLAGS} ${PG_CPPFLAGS}")
```
Comment (akuzm) [inline src/build-defs.cmake:10]: What are you building that needs C++ flags? TimescaleDB itself doesn't use them IIRC. Postgres only uses them for the JIT engine if you configure it with `--with-llvm`

This line in particular doesn't look correct, you can't generally put the C++ flags into the C ones.

Comment (sumerman) [reply src/build-defs.cmake:10]: `pg_config` passes includes in the `CPPFLAGS` on my system it looks like this: `pg_config --cppflags
-I/opt/homebrew/Cellar/icu4c/72.1/include -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -I/opt/homebrew/opt/gettext/include -I/opt/homebrew/opt/openssl@1.1/include -I/opt/homebrew/opt/readline/include -I/opt/homebrew/Cellar/lz4/1.9.4/include` while OpenSSL and MacOSX13.sdk are correctly discovered in by CMake, it knows nothing about `icu4c` and `gettext` leading to #2777

Comment (akuzm) [reply src/build-defs.cmake:10]: I realized that CPPFLAGS are C preprocessor flags, not C++ flags. CMake doesn't have support for them, so it's OK to just paste them with the rest of the C flags. So this line looks correct.

File: src/build-defs.cmake [hunk 2]
```diff
@@ -7,8 +7,8 @@ if(UNIX)
...
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${PG_CPPFLAGS}")
```
Comment (akuzm) [inline src/build-defs.cmake:11]: Maybe we should remove this line at all? We don't use C++ so we don't have to touch the CMAKE_CXX_FLAGS.

File: CMakeLists.txt [hunk 3]
```diff
@@ -361,6 +361,7 @@ get_pg_config(PG_BINDIR --bindir)
 get_pg_config(PG_CFLAGS --cflags)
 get_pg_config(PG_CFLAGS_SL --cflags_sl)
 get_pg_config(PG_CPPFLAGS --cppflags)
+get_pg_config(PG_CFLAGS --cppflags)
```
[the remark this answers is not in the document]
Comment (sumerman) [reply CMakeLists.txt:364]: Apparently it doesn't, but I didn't mean to keep it there. Thanks for the catch!

File: src/build-defs.cmake (hunk 2 above)
Comment (sumerman) [reply src/build-defs.cmake:11]: 🤔

Comment (sumerman) [reply src/build-defs.cmake]: done

**Review by akuzm — APPROVED** [review]

**Review by svenklemm — APPROVED** [review]

**Pull request merged**